samples/maps/demo/cinema-seat-plan/demo.js

```diff
@@ -0,0 +1,184 @@
+                '{#if selected}Selected{else}Available{/if}{/if}'
+        }
+    },
+    title: {
+        text: 'Cinema seat plan',
+        x: 10
+    },
+    subtitle: {
+        text: 'Click to select',
+        x: 10
+    },
+    legend: {
+        enabled: false
+    },
+    mapNavigation: {
+        enabled: false
+    },
+    mapView: {
+        padding: 10,
+        fitToGeometry: {
+            type: 'Polygon',
+            coordinates: [
+                [
+                    [-10, -20],
+                    [325, 170]
+                ]
+            ]
+        }
+    },
+    plotOptions: {
+        series: {
+            dataLabels: {
+                enabled: true,
+                allowOverlap: true,
+                style: {
+                    fontSize: '14px'
+                }
+            },
+            states: {
+                inactive: {
+                    enabled: false
+                }
+            }
+        }
+    },
+    tooltip: {
+        outside: true,
+        formatter: function () {
+            return `Seat <b>${this.point.name}</b> ` +
+                (this.point.selected ? 'Selected' : 'Available');
+        }
+    },
+    responsive: {
+        rules: [{
+            condition: {
+                maxWidth: 450
+            },
+            chartOptions: {
+                series: [{
+                    id: 'seats',
+                    dataLabels: {
+                        style: {
+                            fontSize: '9px'
+                        }
+                    }
+                }, {}]
+            }
+        }]
+    },
+    series: [{
+        id: 'seats',
+        point: {
+            events: {
+                click: function () {
+                    toggleSelection(this.name);
+                    // Update to trigger the accessibility module
+                    this.update({ selected: !this.selected });
+                    this.select(this.selected, true);
+                    // Refresh to update the selection state
+                    this.series.chart.tooltip.refresh(this);
+                }
+            }
+        },
+        mapData: seatPlan,
+        nullColor: '#ccc',
+        borderWidth: 0,
+        states: {
+            select: {
+                color: colors[3]
+            }
+        },
+        // Seats
+        data: (function () {
+            const rot = [18, 14, 8, 5, 0, 0, 0, -5, -8, -14, -18],
+                taken = [
+                    '11000000001', // A1-A11
+                    '10110000000',
+                    '00001100000',
+                    '00000011000',
+                    '00000000110',
+                    '00111000011'
+                ].join('');
+
+            return new Array(66).fill(1).map((seat, i) => [
+                String.fromCharCode(65 + Math.floor(i / 11)) + (i % 11 + 1),
+                +taken[i] ? null : void 0,
+                rot[i % 11]
+            ]);
+        }()),
```

Comment:
This code is quite complex and intimidating. Bearing in mind that the purpose of the demos are to show the public how easy it is to create cool visualizations with Highcharts, I think it is better that we simplify it. Maybe a loop for the rotation is needed, but the array for taken seats I think should be just defined static in the data set.